src/components/ToggleButton/ToggleButton.js

```diff
@@ -5,64 +5,87 @@ import { Button } from 'react-bootstrap';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 
 export const toggleButtonProps = {
-  variants: ['check']
+  variants: ['check', 'circle', 'switch']
 };
 
 export default class ToggleButton extends Component {
   static propTypes = {
     className: PropTypes.string,
     buttonProps: PropTypes.object,
     iconProps: PropTypes.object,
-    initialValue: PropTypes.bool,
+    value: PropTypes.bool,
+    iconOnly: PropTypes.bool,
     variant: PropTypes.oneOf(toggleButtonProps.variants),
     onClick: PropTypes.func
   };
 
   static defaultProps = {
-    initialValue: false,
+    buttonProps: {},
+    iconProps: {},
+    value: false,
     variant: 'check'
   };
 
   constructor(props) {
     super(props);
 
-    this.state = { value: this.props.initialValue };
     this.handleClick = this.handleClick.bind(this);
   }
 
   handleClick(event) {
     const { onClick } = this.props;
 
-    this.setState({ value: !this.state.value });
-
     if (onClick) {
       onClick(event);
     }
   }
 
   render() {
-    const { value } = this.state;
-    const { buttonProps, className, iconProps, variant } = this.props;
+    const {
+      buttonProps,
+      className,
+      iconOnly,
+      iconProps,
+      value,
+      variant
+    } = this.props;
+
+    if (!buttonProps.title) {
+      buttonProps.title = 'Click to toggle';
+    }
 
     let icon = '';
 
     switch (variant) {
+      case 'switch':
+        icon = value ? ['fas', 'toggle-on'] : ['fas', 'toggle-off'];
+        break;
       case 'check':
       default:
         icon = value ? ['fas', 'check-square'] : ['far', 'square'];
         break;
     }
-    const classes = classNames(className, 'btn-toggle');
+    if (iconOnly) {
+      const classes = classNames(className, 'icon-toggle');
+
+      return (
+        <FontAwesomeIcon
+          size="lg"
+          {...iconProps}
+          icon={icon}
+          {...buttonProps}
```

Comment:
This might be unintuitive to a developer unfamiliar with the implementation, is there a need for this spreading to occur?